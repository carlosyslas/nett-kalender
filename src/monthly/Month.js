import React from "react";
import startOfMonth from "date-fns/start_of_month";
import endOfMonth from "date-fns/end_of_month";
import startOfWeek from "date-fns/start_of_week";
import endOfWeek from "date-fns/end_of_week";
import eachDay from "date-fns/each_day";
import format from "date-fns/format";
import isSameMonth from "date-fns/is_same_month";
import { WEEKDAYS } from "./constants";
import DayCard from "./DayCard";
import Header from "./Header";
import "./month.scss";

const getDatesInMonth = date =>
  eachDay(startOfWeek(startOfMonth(date)), endOfWeek(endOfMonth(date)));

const Month = ({
  date: selectedDate,
  events,
  openEditModal,
  openDeleteModal
}) => (
  <div>
    <Header date={selectedDate} />
    <div className="month-view">
      {WEEKDAYS.map(day => (
	<div key={day} className="day-name">
	  {day}
	</div>
      ))}
      {getDatesInMonth(selectedDate).map(date => (
	<DayCard
	  key={format(date, "YYYY-MM-DD")}
	  date={date}
	  isDisabled={!isSameMonth(date, selectedDate)}
	  events={(events && events[format(date, "MM-DD")]) || []}
	  openEditModal={openEditModal}
	  openDeleteModal={openDeleteModal}
	/>
      ))}
    </div>
  </div>
);

export default Month;
