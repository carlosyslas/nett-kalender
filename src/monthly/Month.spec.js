import React from "react";
import { shallow } from "enzyme";
import Month from "./Month";

describe("Month", () => {
  it("renders", () => {
    const events = {
      "08-19": [
	{
	  id: "00841621-bea5-478c-8b33-0576ffd161ba",
	  start: "2019-08-19T07:00:00Z",
	  content: "Deliver coding challenge",
	  color: 1
	}
      ]
    };
    const tree = shallow(
      <Month date={new Date(2018, 7, 19, 0, 0)} events={events} />
    );

    expect(tree.debug()).toMatchSnapshot();
  });
});
