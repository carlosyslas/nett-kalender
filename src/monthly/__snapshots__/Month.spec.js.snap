// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month renders 1`] = `
"<div>
  <Header date={{...}} />
  <div className=\\"month-view\\">
    <div className=\\"day-name\\">
      Sun
    </div>
    <div className=\\"day-name\\">
      Mon
    </div>
    <div className=\\"day-name\\">
      Tue
    </div>
    <div className=\\"day-name\\">
      Wed
    </div>
    <div className=\\"day-name\\">
      Thu
    </div>
    <div className=\\"day-name\\">
      Fri
    </div>
    <div className=\\"day-name\\">
      Sat
    </div>
    <DayCard date={{...}} isDisabled={true} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={true} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={true} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={false} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
    <DayCard date={{...}} isDisabled={true} events={{...}} openEditModal={[undefined]} openDeleteModal={[undefined]} />
  </div>
</div>"
`;
